modules/order/src/Plugin/Field/FieldType/AdjustmentItem.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\commerce_order\Plugin\Field\FieldType;
+
+use Drupal\commerce_order\Adjustment;
+use Drupal\commerce_price\Plugin\Field\FieldType\PriceItem;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\TypedData\DataDefinition;
+
+/**
+ * Plugin implementation of the 'commerce_price' field type.
+ *
+ * @todo See if bojanz throws bricks at me for extended PriceItem.
+ *
+ * @dearBojan: instead of just serializing a Adjustment object, put its
```

Comment:
Having individual columns is definitely more proper. The reason why I wanted to avoid this is because core has no way to maintain the schema, once we choose the columns, we're kinda stuck with them (or we need big migration code). Plus, we need to duplicate validation code between the AdjustmentItem and the Adjustment class. Meanwhile, I'm unsure we have any use of those columns in Views. We'll end up using custom formatters for the whole field anyway.